compiler/rustc_typeck/src/check/gather_locals.rs

```diff
@@ -92,22 +92,22 @@ impl<'a, 'tcx> Visitor<'tcx> for GatherLocalsVisitor<'a, 'tcx> {
     }
 
     fn visit_param(&mut self, param: &'tcx hir::Param<'tcx>) {
-        let old_outermost_fn_param_pat = mem::replace(&mut self.outermost_fn_param_pat, true);
+        let tmp = mem::replace(&mut self.outermost_fn_param_pat, Some(param.ty_span));
```

Comment:
```suggestion
        let old_outermost_fn_param_pat = self.outermost_fn_param_pat.replace(param.ty_span);
```
- `Option` has its own replace method (https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.replace)
- some meaningful name instead of `tmp`